presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestReorderJoins.java

```diff
@@ -78,6 +79,19 @@ public void testEliminateCrossJoinFromTwoJoinBlocks()
                                 anyTree())));
     }
 
+    @Test
+    public void testEliminateCrossJoinWithNonEqualityCondition()
+    {
+        assertPlan("SELECT o.orderkey FROM part p, orders o, lineitem l WHERE p.partkey = l.partkey AND l.orderkey = o.orderkey AND p.partkey <> o.orderkey",
+                anyTree(
+                        join(INNER, ImmutableList.of(aliasPair("orderkey_2", "orderkey")),
+                                anyTree(
+                                        join(INNER, ImmutableList.of(aliasPair("partkey", "partkey_3")),
+                                                anyTree(tableScan("part")),
+                                                anyTree(filter("partkey_3 <> orderkey_2", tableScan("lineitem"))))),
```

Comment:
I don't care about what produces given allias. I want to assert only table join order.